packages/api/src/controllers/resources/v1/mod.rs

```diff
@@ -0,0 +1,125 @@
+use by_axum::{
+    auth::Authorization,
+    axum::{
+        extract::{Path, Query, State},
+        routing::get,
+        Extension, Json, Router,
+    },
+};
+
+use models::{
+    // ResourceDeleteRequest,
+    QueryResponse,
+    Resource,
+    ResourceAction,
+    ResourceByIdAction,
+    ResourceCreateRequest,
+    ResourceQuery,
+    ResourceReadAction,
+    ResourceRepository,
+    ResourceSummary,
+    ResourceUpdateRequest,
+};
+
+#[derive(Clone, Debug)]
+pub struct ResourceConterollerV1 {
+    repo: ResourceRepository,
+}
+
+impl ResourceConterollerV1 {
+    pub fn route(pool: sqlx::Pool<sqlx::Postgres>) -> models::Result<Router> {
+        let repo = Resource::get_repository(pool.clone());
+        let ctrl = Self { repo };
+
+        Ok(Router::new()
+            .route(
+                "/:id",
+                get(Self::get_resource).post(Self::act_resource_by_id),
+            )
+            .route("/", get(Self::list_resources).post(Self::act_resource))
+            .with_state(ctrl))
+    }
+    async fn get_resource(
+        State(ctrl): State<ResourceConterollerV1>,
+        Extension(_auth): Extension<Option<Authorization>>,
+        Path(id): Path<String>,
+    ) -> models::Result<Json<Resource>> {
+        let resource = ctrl
+            .repo
+            .find_one(&ResourceReadAction::new().find_by_id(id))
+            .await?;
+        Ok(Json(resource))
+    }
+
+    async fn list_resources(
+        State(ctrl): State<ResourceConterollerV1>,
+        Extension(_auth): Extension<Option<Authorization>>,
+        Query(params): Query<ResourceQuery>,
+    ) -> models::Result<Json<QueryResponse<ResourceSummary>>> {
+        Ok(Json(ctrl.repo.find(&params).await?))
+    }
```

Comment:
```suggestion
    async fn list_resources(
        State(ctrl): State<ResourceConterollerV1>,
        Extension(_auth): Extension<Option<Authorization>>,
        Query(params): Query<ResourceParam>,
    ) -> models::Result<Json<ResourceGetResponse>> {

        match params {
            ResourceParam::Query(q) => Ok(Json(ResourceGetResponse::Query(ctrl.repo.find(&params).await?))),
            ResourceParam::Read(q) => Ok(Json(ResourceGetResponse::Read(ctrl.repo.find_one(&params).await?))),
        }
        
    }
```

이런 패턴을 쓰길 권장합니다.

read_action 으로 정의된 것은 find_one 에 사용할 수 있고,
query_action, queryable 로 정의된 것은 find에 사용할 수 있어요.

find 는 지금처럼 QueryResponse<..Summary> 를 리턴하고,
find_one(read_action)은 Resource 와 같은 하나의 전체 모델을 리턴하는데요.

..GetResponse 는 두개를 수용할 수 있게 되어 있습니다.